File: App.Application/Features/Global/BankTransactions/Queries/GetAllCombinedBankTransactions/CombinedBankTransactionVM.cs
namespace OOH.Application.Features.Global.BankTransactions.Queries.GetAllCombinedBankTransactions
{
    public class CombinedBankTransactionVM
    {
        public string ApprovalId { get; set; }
        public string ApprovalName { get; set; }
        public decimal Amount { get; set; }
        public string FromBankName { get; set; }
        public string ToBankName { get; set; }
        public string CompletedOn { get; set; }
        public decimal? RunningBalanceBank1 { get; set; } // Running balance for FromBank (Debit)
        public decimal? RunningBalanceBank2 { get; set; } // Running balance for ToBank (Credit)
    }
}

File: app.client/src/api/models/CombinedBankTransactionVM.ts
export interface CombinedBankTransactionVM {
    approvalId: string;
    approvalName: string;
    approvalReference?: string;
    amount: number;
    fromBankName: string | null;
    toBankName: string | null;
    completedOn: string;
    approvalType?: string;
    runningBalanceBank1: number | null;
    runningBalanceBank2: number | null;
}
